### ISSUE TITLE
Caddy HTTPS certificate

### ISSUE THREAD

**Issue by ssonparote** [sghctoma/sst#25]
Hello, 

Unfortunately I'm more skilled on the hardware side than software (which looks impressive!)

I'm trying to run it in Docker desktop, and managed to get a successful build, however when trying to go to localhost:443, I get an "Client sent an HTTP request to an HTTPS server." error, or when using https://localhost:443, I get a "This site can’t provide a secure connection" - it seems like the default caddyfile is not providing a self-signed certificate. Is there a quick fix, or should I just try to build it on a web platform?

**Comment by sghctoma** [comment]
Hello,

I've just reproduced this issue, but it's not because of Caddy. The dashboard uses an older version of Flask, and a recent update to one of Flask's dependencies is no longer compatible with that older version. I'll try to release a new version today with updated Python components.

In the meantime, could you please check your dashboard container's logs just to be sure you have the same problem? If it's the same problem, there should be a bunch of "ImportError: cannot import name 'url_quote' from 'werkzeug.urls'" errors.

![Capture](https://github.com/sghctoma/sst/assets/214533/2c7bb832-a538-4d7e-be5f-bc50d2d9c54b)

**Comment by sghctoma** [comment]
Ah, you are right, the problem still persist with a properly running dashboard. I did not catch that when I tried, because I was accessing the server from another machine via its IP address, and the problem is with the hostname. The default Caddyfile has only one hostname configured (_sst.caddy.local_) so when you try to access the server on _localhost_, SNI won't accept it. A quick check to confirm this would be trying https://127.0.0.1 instead of https://localhost.

To fix this, both occurances of _sst.caddy.local_ could be replaced with _localhost_, or one could simply append _localhost_ to the host names like this:

```
{
	admin off
	default_sni sst.caddy.local
}
:443, sst.caddy.local, localhost {
	tls internal
	handle {
		reverse_proxy dashboard:5000
	}
	log {
		output file /data/caddy-access.log
		format json
	}
}
```
